spec/services/photo_service_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+describe PhotoService do
+  before do
+    alice_eve_spec = alice.aspects.create(name: "eve aspect")
+    alice.share_with(eve.person, alice_eve_spec)
+    alice_bob_spec = alice.aspects.create(name: "bob aspect")
+    alice.share_with(bob.person, alice_bob_spec)
+    @alice_eve_photo = alice.post(:photo, pending: false, user_file: File.open(photo_fixture_name),
+                                 to: alice_eve_spec.id)
+    @alice_bob_photo = alice.post(:photo, pending: false, user_file: File.open(photo_fixture_name),
+                                 to: alice_bob_spec.id)
+    @alice_public_photo = alice.post(:photo, pending: false, user_file: File.open(photo_fixture_name), to: "all")
+    @bob_photo1 = bob.post(:photo, pending: true, user_file: File.open(photo_fixture_name), to: "all")
+  end
+
+  describe "visible_photo" do
+    it "returns a user's photo" do
+      photo = photo_service.visible_photo(@bob_photo1.guid)
+      expect(photo.guid).to eq(@bob_photo1.guid)
+    end
+
+    it "returns another user's public photo" do
+      photo = photo_service.visible_photo(@alice_public_photo.guid)
+      expect(photo.guid).to eq(@alice_public_photo.guid)
+    end
+
+    it "returns another user's shared photo" do
+      photo = photo_service.visible_photo(@alice_bob_photo.guid)
+      expect(photo.guid).to eq(@alice_bob_photo.guid)
+    end
+
+    it "returns nil for other user's private photo" do
+      photo = photo_service.visible_photo(@alice_eve_photo.guid)
+      expect(photo).to be_nil
```

Comment:
I don't see why it's different here. Both Eve and Bob shares with Alice. In other tests, bob don't share anything with Alice

Reply:
It was a photo that Alice shared with Eve not Bob so when bob tries to pull it because he somehow got the GUID it shouldn't return it.  The previous test was when that Alice shared with Bob in which case it should work.